Commit 093cc85 break compilation in debug mode

After the commit 093cc8507c0dc74f24644a3a66afed94a05a500d (fix compilation warning), the code no more compile in debug mode.
This is due to missing `;`in the use of the macro `CORE_UNUSED ` in file KeyFrameValueInterpolators.hpp.
Two things : 
  - this macro is quite complicated to just remove a warning and should be simplified to `#define CORE_UNUSED (x) void(x)`
  - Adding a `;` might generates another warning in release mode as the macro does not expend the same way in release.

(not) Compiled on MacOs with gcc 9.3.0
```
Radium-Engine/src/Core/Animation/KeyFramedValueInterpolators.hpp: In function 'T Ra::Core::Animation::linearInterpolate(const Ra::Core::Animation::KeyFramedValue<T>&, Scalar) [with T = bool; Scalar = float]':
Radium-Engine/src/Core/CoreMacros.hpp:137:24: error: expected ';' before 'do'
  137 | #   define MACRO_START do {
      |                        ^~
Radium-Engine/src/Core/CoreMacros.hpp:146:5: note: in expansion of macro 'MACRO_START'
  146 |     MACRO_START        \
      |     ^~~~~~~~~~~
Radium-Engine/src/Core/Animation/KeyFramedValueInterpolators.hpp:32:5: note: in expansion of macro 'CORE_UNUSED'
   32 |     CORE_UNUSED( dt )
      |     ^~~~~~~~~~~
```

--- a/src/Core/Animation/KeyFramedValueInterpolators.hpp
+++ b/src/Core/Animation/KeyFramedValueInterpolators.hpp
@@ -28,8 +28,8 @@ template <>
 inline bool linearInterpolate<bool>( const KeyFramedValue<bool>& keyframes, const Scalar t ) {
     CORE_ASSERT( keyframes.size() > 0, "Keyframe vectors must contain at least one keyframe." );
     auto [i, j, dt] = keyframes.findRange( t );
-    CORE_UNUSED( j )
-    CORE_UNUSED( dt )
+    CORE_UNUSED( j );
+    CORE_UNUSED( dt );
     return keyframes[i].second;
 }
 
@@ -38,8 +38,8 @@ template <>
 inline int linearInterpolate<int>( const KeyFramedValue<int>& keyframes, const Scalar t ) {
     CORE_ASSERT( keyframes.size() > 0, "Keyframe vectors must contain at least one keyframe." );
     auto [i, j, dt] = keyframes.findRange( t );
-    CORE_UNUSED( j )
-    CORE_UNUSED( dt )
+    CORE_UNUSED( j );
+    CORE_UNUSED( dt );
     return keyframes[i].second;
 }
 

